io/src/main/scala/fs2/io/io.scala

```diff
@@ -124,4 +124,30 @@ package object io {
     */
   def toInputStream[F[_]](implicit F: Effect[F], ec: ExecutionContext): Pipe[F,Byte,InputStream] =
     JavaInputOutputStream.toInputStream
+
+  /**
+    * Reads all the values of the iterator into a Stream.
+    */
+  def readIterator[F[_], A](iterator: Iterator[A])(implicit F: Effect[F]): Stream[F, A] = {
```

Comment:
I think we can move this method to `Stream` companion as `Stream.fromIterator` -- the effect constraint can be loosened to `Sync`.

Reply:
You can make this a bit more concise with:
```scala
  def readIterator2[F[_], A](iterator: Iterator[A])(implicit F: Sync[F]): Stream[F, A] =
    Stream.unfoldEval(iterator)(i => F.delay(i.hasNext.guard[Option].as(i.next -> i)))
```

it's a style choice though, so up to you